salt/minion.py

```diff
@@ -1532,37 +1548,13 @@ def _handle_decoded_payload(self, data):
         # python needs to be able to reconstruct the reference on the other
         # side.
         instance = self
-        multiprocessing_enabled = self.opts.get('multiprocessing', True)
-        if multiprocessing_enabled:
-            if sys.platform.startswith('win'):
-                # let python reconstruct the minion on the other side if we're
-                # running on windows
-                instance = None
-            with default_signals(signal.SIGINT, signal.SIGTERM):
-                process = SignalHandlingMultiprocessingProcess(
-                    target=self._target, args=(instance, self.opts, data, self.connected)
-                )
+        if self.opts.get('ipc_mode', '') == 'tcp':
+            pull_uri = int(self.opts.get('tcp_minion_workers', 4516))
         else:
-            process = threading.Thread(
-                target=self._target,
-                args=(instance, self.opts, data, self.connected),
-                name=data['jid']
-            )
-
-        if multiprocessing_enabled:
-            with default_signals(signal.SIGINT, signal.SIGTERM):
-                # Reset current signals before starting the process in
-                # order not to inherit the current signal handlers
-                process.start()
-        else:
-            process.start()
-
-        # TODO: remove the windows specific check?
-        if multiprocessing_enabled and not salt.utils.platform.is_windows():
-            # we only want to join() immediately if we are daemonizing a process
-            process.join()
-        else:
-            self.win_proc.append(process)
+            pull_uri = os.path.join(self.opts['sock_dir'], 'worker_pull.ipc')
+        log.error("CLINT SEND %s", data)
+        client = salt.transport.ipc.IPCMessageClient(pull_uri)
```

Comment:
Is it really necessary to instantiate a new client on every receipt of a payload? Couldn't this just be reused?